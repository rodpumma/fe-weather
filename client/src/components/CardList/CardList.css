.weather-card__day-header {
  font-weight: 600;
  margin: 8px 0;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  width: 130px;
}