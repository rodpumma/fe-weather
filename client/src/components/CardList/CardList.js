import React from 'react'
import Card from '../card/Card'
import './CardList.css'
import Title from '../Title/Title';

const CardList = ({cards, name = null}) => {
  return (
    <>
      {name ? <Title text={name} />
        : null}
      <div className="weather-list__content" >
        {cards.map((data, i) => (
          <div className="weather-list__element" key={i}>
            {data.name ? <Title text={data.name} />
              : null}
            <Card info={data} />
          </div>
        ))}
      </div>
    </>
  )
}

export default CardList