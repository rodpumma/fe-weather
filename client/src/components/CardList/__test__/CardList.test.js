import React from 'react'
import CardList from '../CardList'

import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import renderer from 'react-test-renderer'

afterEach(cleanup)
describe('<CardList />', () => {
  const props = {
    cards: [
      {
        name: 'test card name one',
        dt_txt: '2020-09-26',
        main: {
          temp_min: 12.8,
          temp_max: 17,
        }
      },
      {
        name: 'test card name two',
        dt_txt: '2020-09-26',
        main: {
          temp_min: 12.8,
          temp_max: 17,
        }
      },
    ]
  }

  it('Check <CardList /> render', () => {
    const { container } = render(
      <CardList {...props} />,
    )
    expect(container.querySelectorAll('Title')).toHaveLength(0)
    expect(container.querySelector('Title'))
      .toBeNull()
  })

  it('<CardList /> matches snapshop', () => {
    const tree = renderer
      .create(<CardList {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
})
