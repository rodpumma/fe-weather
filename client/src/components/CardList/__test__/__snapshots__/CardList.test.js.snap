// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardList /> <CardList /> matches snapshop 1`] = `
<div
  className="weather-list__content"
>
  <div
    className="weather-list__element"
  >
    <pre
      className="weather-card__day-header"
      data-testid="title"
    >
      test card name one
    </pre>
    <div
      className="weather-card__content"
    >
      <div
        className="weather-card__card"
      >
        <span
          className="weather-card__day-title"
          data-testid="day-title"
        >
          vienes
        </span>
        <img
          alt="Weather icon"
          className="weather-card__icon"
          src="weather-image.png"
        />
        <div
          className="weather-card__temp-content"
        >
          <pre
            className="weather-card__temp-label"
          >
            min
          </pre>
          <span
            className="weather-card__min-temp"
            data-testid="min-temp"
          >
            12
            째
          </span>
          <pre
            className="weather-card__temp-label"
          >
            max
          </pre>
          <span
            data-testid="max-temp"
          >
            17
            째
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="weather-list__element"
  >
    <pre
      className="weather-card__day-header"
      data-testid="title"
    >
      test card name two
    </pre>
    <div
      className="weather-card__content"
    >
      <div
        className="weather-card__card"
      >
        <span
          className="weather-card__day-title"
          data-testid="day-title"
        >
          vienes
        </span>
        <img
          alt="Weather icon"
          className="weather-card__icon"
          src="weather-image.png"
        />
        <div
          className="weather-card__temp-content"
        >
          <pre
            className="weather-card__temp-label"
          >
            min
          </pre>
          <span
            className="weather-card__min-temp"
            data-testid="min-temp"
          >
            12
            째
          </span>
          <pre
            className="weather-card__temp-label"
          >
            max
          </pre>
          <span
            data-testid="max-temp"
          >
            17
            째
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
